cli/src/main/scala/com/codacy/analysis/cli/Main.scala

```diff
@@ -37,51 +37,70 @@ class MainImpl extends CLIApp {
   def runCommand(command: Command): Int = {
     command match {
       case analyse: Analyse =>
-        cleanup(analyse.directory)
-        Logger.setLevel(analyse.options.verboseValue)
-
-        val formatter: Formatter = Formatter(analyse.format, analyse.output)
-        val analyser: Analyser[Try] = Analyser(analyse.extras.analyser)
-        val fileCollector: FileCollector[Try] = FileCollector.defaultCollector()
-        val environment = new Environment(sys.env)
-        val codacyClientOpt: Option[CodacyClient] = Credentials.get(environment, analyse.api).map(CodacyClient.apply)
-
-        val remoteProjectConfiguration: Either[String, ProjectConfiguration] = codacyClientOpt.fold {
-          "No credentials found.".asLeft[ProjectConfiguration]
-        } { _.getRemoteConfiguration }
-
-        val analysisResults = new AnalyseExecutor(
-          analyse.tool,
-          analyse.directory,
-          formatter,
-          analyser,
-          fileCollector,
-          remoteProjectConfiguration,
-          analyse.parallel,
-          analyse.allowNetworkValue,
-          analyse.forceFilePermissionsValue,
-          analyse.toolTimeout).run()
-
-        val uploadResultFut = uploadResults(codacyClientOpt)(analyse.uploadValue, analyse.commitUuid, analysisResults)
-
-        val uploadResult = if (analyse.uploadValue) {
-          Try(Await.result(uploadResultFut, Duration.Inf)) match {
-            case Failure(err) =>
-              logger.error(err.getMessage)
-              Left(err.getMessage)
-            case Success(Left(err)) =>
-              logger.warn(err)
-              Left(err)
-            case Success(Right(_)) =>
-              logger.info("Completed upload of results to API")
-              Right(())
-          }
-        } else Right(())
-
-        new ExitStatus(analyse.maxAllowedIssues, analyse.failIfIncompleteValue).exitCode(analysisResults, uploadResult)
+        if (isValidTool(analyse.tool)) {
```

Comment:
I think this kind of validation would make more sense inside the logic where we search the tools. 
Adding it here just seems to add logic in a `controller` where should not be any.

Reply:
At least we should move this to the executor